server/src/controllers/eventController.ts

```diff
@@ -333,112 +305,692 @@ eventController.get('/:id', async (req: Request, res: Response) => {
+        if (!updatedEvent) {
+          return res.status(400).json({error: `Event ${id} not found`});
+        }
+        return res.json(updatedEvent);
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          res.status(400).json({error: error.message});
+          return;
+        }
+        if (error instanceof Prisma.PrismaClientValidationError) {
+          res.status(400).json({error: error.message});
+          return;
+        }
+        res.status(500).json({error: 'Internal Server Error'});
+      }
+    },
+);
+
+/**
+ * @swagger
+ * /2/events/{id}:
+ *   delete:
+ *     summary: delete an event
+ *     tags:
+ *     - New Event API
+ *     parameters:
+ *     - $ref: '#/components/parameters/id'
+ *     security:
+ *       - bearerAuth: []
+ *     responses:
+ *       204:
+ *         description: deleted successfully.
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+eventController.delete('/:id', async (req: Request, res: Response) => {
+  try {
+    const id = req.params.id;
+    const deletedEvent = await prisma.events.softDelete(Number(id));
+    if (!deletedEvent) {
+      return res.status(400).json({error: `Event ${id} not found`});
+    }
+    return res.status(204).json('Event successfully deleted');
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    if (error instanceof Prisma.PrismaClientValidationError) {
+      res.status(400).json({error: error.message});
+      return;
+    }
+    res.status(500).json({error: 'Internal Server Error'});
+  }
+});
+
+
+/**
+ * @swagger
+ * /2/events/checkin:
+ *   put:
+ *     summary: update eventticket status
+ *     tags:
+ *     - New Event API
+ *     security:
+ *       - bearerAuth: []
+ *     requestBody:
+ *       description: event ticket status
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/components/requestBodies/CheckIn'
+ *     responses:
+ *       200:
+ *         description: event ticket updated successfully.
+ *       400:
+ *         description: bad request
+ *         content:
+ *           application/json:
+ *             schema:
+ *              $ref: '#/components/schemas/Error'
+ *       422:
+ *         description: invalid input
+ *         content:
+ *           application/json:
+ *             schema:
+ *              $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error. An error occurred while processing the request.
+ */
+eventController.put('/checkin', async (req: Request, res: Response) => {
+  try {
+    const {ticketID, isCheckedIn}= req.body;
```

Comment:
Super tiny, but should be a space before the `=`

Reply:
fixed